GG/src/ListBox.cpp

```diff
@@ -431,12 +443,18 @@ void ListBox::Row::SetColAlignments(const std::vector<Alignment>& aligns)
     if (aligns == m_col_alignments)
         return;
 
-    m_col_alignments = aligns;
+    m_col_alignments.resize(m_cells.size(), ALIGN_NONE);
+
+    for (std::size_t i = 0; i < m_col_alignments.size(); ++i) {
+        if (i < aligns.size())
+            m_col_alignments.at(i) = aligns.at(i);
+    }
+
     auto&& layout = GetLayout();
     ValidateLayoutSize(layout.get(), aligns.size());
     for (std::size_t ii = 0; ii < m_cells.size(); ++ii) {
-        if (m_cells[ii])
-            layout->SetChildAlignment(m_cells[ii].get(), m_row_alignment | m_col_alignments[ii]);
+        if (m_cells.at(ii))
```

Comment:
`ii` ranges from 0 to `m_cells.size() - 1`, so using `m_cells.at(ii)` here instead of `m_cells[ii]` seems pointless.